src/map/clif.c

```diff
@@ -20226,6 +20226,189 @@ void clif_skill_scale(struct block_list *bl, int src_id, int x, int y, uint16 sk
 #endif
 }
 
+void clif_stylist_vector_init(void)
+{
+	int i;
+	for (i = 0; i < MAX_STYLIST_TYPE; i++) {
+		VECTOR_INIT(stylist_data[i]);
+	}
+}
+
+void clif_stylist_vector_clear(void)
+{
+	int i;
+	for (i = 0; i < MAX_STYLIST_TYPE; i++) {
+		VECTOR_CLEAR(stylist_data[i]);
+	}
+}
+
+bool clif_stylist_read_db_libconfig(void)
+{
+	struct config_t stylist_conf;
+	struct config_setting_t *stylist = NULL, *it = NULL;
+	const char *config_filename = "db/stylist_db.conf"; // FIXME hardcoded name
+	int i = 0;
+
+	if (!libconfig->load_file(&stylist_conf, config_filename))
+		return false;
+
+	if ((stylist = libconfig->setting_get_member(stylist_conf.root, "stylist_db")) == NULL) {
+		ShowError("can't read %s\n", config_filename);
+		return false;
+	}
+
+	clif->stylist_vector_clear();
+
+	while ((it = libconfig->setting_get_elem(stylist, i++))) {
+		clif->stylist_read_db_libconfig_sub(it, i - 1, config_filename);
+	}
+
+	libconfig->destroy(&stylist_conf);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", i, config_filename);
+	return true;
+}
+
+bool clif_stylist_read_db_libconfig_sub(struct config_setting_t *it, int idx, const char *source)
+{
+	struct stylist_data_entry entry = { 0 };
+	int i32 = 0, type = 0;
+	int64 i64 = 0;
+
+	nullpo_ret(it);
+	nullpo_ret(source);
+
+	if (!itemdb->lookup_const(it, "Type", &type) || type >= MAX_STYLIST_TYPE || type < 0) {
+		ShowWarning("clif_stylist_read_db_libconfig_sub: Invalid or missing Type (%d) in \"%s\", entry #%d, skipping.\n", type, source, idx);
+		return false;
+	}
+	if (!itemdb->lookup_const(it, "Id", &i32) || i32 <= 0) {
+		ShowWarning("clif_stylist_read_db_libconfig_sub: Invalid or missing Id (%d) in \"%s\", entry #%d, skipping.\n", i32, source, idx);
+		return false;
+	}
+	entry.id = i32;
+
+	if (libconfig->setting_lookup_int64(it, "Zeny", &i64)) {
+		if (i64 > MAX_ZENY) {
+			ShowWarning("clif_stylist_read_db_libconfig_sub: zeny is too big in \"%s\", entry #%d, capping to MAX_ZENY.\n", source, idx);
+			entry.zeny = MAX_ZENY;
+		} else {
+			entry.zeny = (int)i64;
+		}
+	}
+
+	if (itemdb->lookup_const(it, "ItemID", &i32))
+		entry.itemid = i32;
+
+	if (itemdb->lookup_const(it, "BoxItemID", &i32))
+		entry.boxid = i32;
+
+	VECTOR_ENSURE(stylist_data[type], 1, 1);
+	VECTOR_PUSH(stylist_data[type], entry);
+	return true;
+}
+
+bool clif_style_change_validate_requirements(struct map_session_data *sd, int type, uint16 idx)
+{
+	struct item it;
+	struct stylist_data_entry *entry;
+
+	nullpo_retr(false, sd);
+	Assert_retr(false, type >= 0 && type < MAX_STYLIST_TYPE);
+	Assert_retr(false, idx <= VECTOR_LENGTH(stylist_data[type]));
```

Comment:
should be ``idx < VECTOR_LENGTH(stylist_data[type]))``

Reply:
`VECTOR_LENGTH(stylist_data[type]))` is valid index in here

Reply:
idx used as vector index. index must be from 0 to len-1. because if use index with len, it will be one item outside of array.

Reply:
ah yes my mistake, i was using the index from the packet directly which starts from `1 to len`, fixed.